crates/ruff_linter/src/rules/flake8_pyi/rules/custom_type_var_return_type.rs

```diff
@@ -79,46 +81,51 @@ impl Violation for CustomTypeVarReturnType {
 
 /// PYI019
 pub(crate) fn custom_type_var_return_type(
-    checker: &mut Checker,
-    function_def: &ast::StmtFunctionDef,
-) {
+    checker: &Checker,
+    binding: &Binding,
+) -> Option<Diagnostic> {
+    if !matches!(binding.kind, BindingKind::FunctionDefinition(_)) {
+        return None;
+    }
```

Comment:
I think this is redundant considering you have `let function_def = binding.statement(semantic)?.as_function_def_stmt()?;` a few lines below. That already ensures that we won't emit this diagnostic on a `Binding` unless that binding is associated with a function-definition statement

```suggestion
```